### PR TITLE
add explaination about `nested-loop-join-cache-capacity`

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

add explaination about a new system variable `nested-loop-join-cache-capacity` to `tidb-configuration-file.md`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3703
- Other reference link(s):

### CODE REVIEW

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -355,6 +355,12 @@ Configuration items related to performance.
 - Default: `NO_PRIORITY`
 - Optional values: `NO_PRIORITY`, `LOW_PRIORITY`, `HIGH_PRIORITY` and `DELAYED`.
 
+### `nested-loop-join-cache-capacity`
+
++ The upper memory limit for the Least Recently Used (LRU) nested loop join cache, which indicates the upper memory threshold (in bytes).
```
Comment (yikeke) [inline tidb-configuration-file.md:360]:
```suggestion
+ The maximum memory usage for the Least Recently Used (LRU) algorithm of the nested loop join cache (in bytes).
```
`which indicates` part seems to carry no more information than the main clause, so I suggest removing it. Please help confirm whether my suggestion is technically correct. (Is LRU is an algorithm?) @qw4990

**Comment by TomShawn** [comment]
@yikeke PTAL

**Pull request merged**